app/src/main/java/com/example/shiv/mdbpokedexproject2/MainActivity.java

```diff
@@ -0,0 +1,282 @@
+    RecyclerView rview;
+    PokedexAdapter pokedexAdapter;
+
+    boolean isGridView = false;
+
+    //variables for the filter method
+    public static int min_attack = 0;
+    public static int min_defense = 0;
+    public static int min_health = 0;
+    public static ArrayList<String> type = new ArrayList<>();
+    public static boolean displayAllTypes = true;
+
+    //list of filtered pokemons to display
+    private List<Pokedex.Pokemon> mModels = new ArrayList<>();
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        //Initialize
+        rview = (RecyclerView) findViewById(R.id.recyclableView);
+        rview.setLayoutManager(new LinearLayoutManager(getApplicationContext()));
+        Toolbar myToolbar = (Toolbar) findViewById(R.id.my_toolbar);
+        setSupportActionBar(myToolbar);
+
+        //Arraylist of all the pokemon
+        ArrayList <Pokedex.Pokemon> pokedex = new Pokedex().getPokemon();
+
+        //Since the pokedex types aren't working, initialize random types for proof of concept
+        for (int i = 0; i < 30; i++) {
+            pokedex.get(i).type = "Dragon";
+        }
+        for (int i = 31; i < 60; i++) {
+            pokedex.get(i).type = "Fairy";
+        }
+        for (int i = 61; i < 90; i++) {
+            pokedex.get(i).type = "Normal";
+        }
+        for (int i = 91; i < pokedex.size(); i++) {
+            pokedex.get(i).type = "Fire";
+        }
+
+
+
+        pokedexAdapter = new PokedexAdapter(getApplicationContext(), pokedex, new Comparator<Pokedex.Pokemon>() {
+            @Override
+            public int compare(Pokedex.Pokemon pokemon, Pokedex.Pokemon t1) {
+                return pokemon.name.compareTo(t1.name);
+            }
+        });
+
+        rview.setAdapter(pokedexAdapter);
+
+        //update mModels with all the pokemon to display on create
+        mModels = new ArrayList<>();
+        for (Pokedex.Pokemon k: pokedex) {
+            mModels.add(k);
+        }
+
+        pokedexAdapter.add(mModels);
+
+    }
+
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        MenuInflater inflater = getMenuInflater();
+        inflater.inflate(R.menu.search_bar, menu);
+
+        // Associate searchable configuration with the SearchView
+        SearchManager searchManager =
+                (SearchManager) getSystemService(Context.SEARCH_SERVICE);
+        SearchView searchView =
+                (SearchView) menu.findItem(R.id.search).getActionView();
+        searchView.setSearchableInfo(
+                searchManager.getSearchableInfo(getComponentName()));
+        searchView.setOnQueryTextListener(this);
+
+
+
+        return true;
+
+    }
+
+    //Implement active search bar listening
+    @Override
+    public boolean onQueryTextChange(String query) {
+
+        //Case 1: If the query is an integer
+        try {
```

Comment:
generally not safe to use a try catch as an if statement (what if there was some other error)